### PR TITLE
Auto config OidcReactiveOAuth2UserService When Oauth2UserService bean is Present

### PR DESCRIPTION
**Pull request by kse-music**
Closes gh-15846

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java [hunk 1]
```diff
@@ -4506,7 +4506,9 @@ private ReactiveOAuth2UserService<OidcUserRequest, OidcUser> getOidcUserService(
 			if (bean != null) {
 				return bean;
 			}
-			return new OidcReactiveOAuth2UserService();
+			OidcReactiveOAuth2UserService reactiveOAuth2UserService = new OidcReactiveOAuth2UserService();
```
Comment (jzheaux) [inline config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java:4509]: Will you please add a test that demonstrates that when there is an `ReactiveOauth2UserService` bean, this OIDC service uses it correctly?

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @kse-music! Please see my feedback inline.

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java (hunk 1 above)
Comment (kse-music) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java:4509]: Done

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Thanks, @kse-music! This is now merged into `main`.

**Pull request merged**